File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/addit/PayloadAdapter.java
package com.adadapted.android.sdk.core.addit;

import com.adadapted.android.sdk.core.device.DeviceInfo;

import java.util.List;

public interface PayloadAdapter {
    interface Callback {
        void onSuccess(List<AdditContent> content);
    }

    void pickup(DeviceInfo deviceInfo, Callback callback);
    void publishEvent(PayloadEvent event);
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/addit/PayloadAdapter.kt
package com.adadapted.android.sdk.core.addit

import com.adadapted.android.sdk.core.device.DeviceInfo

interface PayloadAdapter {
    interface Callback {
        fun onSuccess(content: List<AdditContent>)
    }

    fun pickup(deviceInfo: DeviceInfo, callback: Callback)
    fun publishEvent(event: PayloadEvent)
}